libs/inform/src/lib/abstracts/modal/modal.abstract.component.ts

```diff
@@ -4,7 +4,10 @@ import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/c
  * An abstract for the NgxModalService
  */
 @Directive()
-export class NgxModalAbstractComponent<ActionType extends string = string, DataType = any> {
+export class NgxModalAbstractComponent<
+	ActionType extends string | { type: string; data: any },
```

Comment:
I would turn `string | {type: string; data: any}` into a type you can just share. 

Something like this:
```
export type NgxModalActionType<StringType extends string = string, DataType = any> = StringType | {type: StringType; data: DataType}
```